File: programmers/cpp/programmers_greedy_1.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
	int answer = 0;
	vector<int > v;

	for (int i = 0; i < reserve.size(); i++)
	{
		for (int j = 0; j < lost.size(); j++)
		{
			if (reserve[i] == lost[j])
			{
				lost.erase(lost.begin() + j);
				v.push_back(reserve[i]);
				break;
			}
		}
	}
	for (int i = 0; i < v.size(); i++)
	{
		auto it = find(reserve.begin(), reserve.end(), v[i]);
		if (it != reserve.end())
			reserve.erase(it);
	}
	for (int i = 0; i < reserve.size(); i++)
	{
		for (int j = 0; j < lost.size(); j++)
		{
			if (reserve[i] - 1 == lost[j] || reserve[i] + 1 == lost[j])
			{
				lost.erase(lost.begin() + j);
				break;
			}

		}
	}
	answer = n - lost.size();

	return answer;
}

File: programmers/python/programmers_greedy_1.py
def solution(n, lost, reserve):
	l = [1] * (n + 2)
	for i in lost:
		l[i] -= 1
	for i in reserve:
		l[i] += 1
	for i in range(1, n + 1):
		if l[i] == 2:
			if l[i - 1] == 0:
				l[i - 1] = 1
				l[i] = 1
			elif l[i + 1] == 0:
				l[i + 1] = 1
				l[i] = 1
	answer = 0
	for i in range(1, n + 1):
		answer += 1 if l[i] > 0 else 0
	return answer